src/main/java/com/epam/ta/reportportal/dao/WidgetContentRepositoryImpl.java

```diff
@@ -757,9 +757,9 @@ public PassingRateStatisticsResult summaryPassingRateStatistics(Filter filter, S
 	}
 
 	@Override
-	public List<MostTimeConsumingTestCasesContent> mostTimeConsumingTestCasesStatistics(Filter filter) {
+	public List<MostTimeConsumingTestCasesContent> mostTimeConsumingTestCasesStatistics(Filter filter, int limit) {
 		return dsl.with(ITEMS)
-				.as(QueryBuilder.newBuilder(filter, collectJoinFields(filter, Sort.unsorted())).build())
+				.as(QueryBuilder.newBuilder(filter).with(limit).build())
```

Comment:
why is collectJoinFields removed?

Reply:
Most time consuming widget uses **TestItem** Filter Target, so collect join creates redundant join statements